acquia_lift.admin.wizard.inc

```diff
@@ -923,15 +932,19 @@ function acquia_lift_personalize_campaign_wizard_variations_submit(&$form, &$for
     switch ($form_state['values']['variations']['add_variation']['option_set_types']['option_set_type']) {
       case "block":
         $submitBlockHandler($values['variations']['add_variation']['details']['block']['content']);
+        $requires_cleanup = TRUE;
         break;
       case "element":
         break;
     }
   }
-  // Clear out any redirects set by option set forms.
-  unset($form_state['redirect']);
-  // Set the clicked button ID back.
-  $form_state['clicked_button']['#id'] = $click_id;
+
+  if ($requires_cleanup) {
+    // Clear out any redirects set by option set forms.
+    unset($form_state['redirect']);
+    // Set the clicked button ID back.
+    $form_state['clicked_button']['#id'] = $click_id;
+  }
```

Comment:
Yeah - I agree.  Each time I see this code it bugs me just a little bit more.  I can fit it in this sprint as a follow up in the workflow branches.